coding/reader.hpp

```diff
@@ -19,10 +20,12 @@ class Reader
   DECLARE_EXCEPTION(SizeException, Exception);
   DECLARE_EXCEPTION(ReadException, Exception);
 
+  using TReaderPtr = unique_ptr<Reader>;
```

Comment:
Writing unique_ptr<Reader> every time if too ugly for me.
I hope `using TReaderUniquePtr = unique_ptr<Reader>` makes both of us satisfied.